saleor/graphql/core/utils/__init__.py

```diff
@@ -62,3 +68,32 @@ def get_node_optimized(qs, lookup, info):
     qs = qs.filter(**lookup)
     qs = gql_optimizer.query(qs, info)
     return qs[0] if qs else None
+
+
+def validate_slug_and_generate_if_needed(
+    instance: Type["Model"],
+    slugable_field: str,
+    cleaned_input: dict,
+    slug_field_name: str = "slug",
+) -> dict:
+    """Validate slug from input and generate in create mutation if is not given."""
+    # update mutation - just check if value is not empty
+    if getattr(instance, slug_field_name):
```

Comment:
If `slug_field_name` is not present in the `instance` this would raise an error. Don't we want to prevent from that by using something like `getattr(instance, slug_field_name, None)`?

Reply:
Maybe the best solution will be checking if `instance` has `pk` value instead of checking `slug_field_name`, because `pk` is always present in model instance.